pkg/csv2lp/csvTable.go

```diff
@@ -0,0 +1,464 @@
+}
+
+// AppendLine appends a protocol line to the supplied buffer and returns appended buffer or an error if any
+func (t *CsvTable) AppendLine(buffer []byte, row []string) ([]byte, error) {
+	if t.computeLineProtocolColumns() {
+		// validate column data types
+		if t.cachedFieldValue != nil && !IsTypeSupported(t.cachedFieldValue.DataType) {
+			return buffer, CsvColumnError{
+				t.cachedFieldValue.Label,
+				fmt.Errorf("data type '%s' is not supported", t.cachedFieldValue.DataType),
+			}
+		}
+		for _, c := range t.cachedFields {
+			if !IsTypeSupported(c.DataType) {
+				return buffer, CsvColumnError{
+					c.Label,
+					fmt.Errorf("data type '%s' is not supported", c.DataType),
+				}
+			}
+		}
+	}
+
+	if t.cachedMeasurement == nil {
+		return buffer, errors.New("no measurement column found")
+	}
+	measurement := t.cachedMeasurement.Value(row)
+	if measurement == "" {
+		return buffer, CsvColumnError{
+			t.cachedMeasurement.Label,
+			errors.New("no measurement supplied"),
+		}
+	}
+	buffer = append(buffer, escapeMeasurement(measurement)...)
+	for _, tag := range t.cachedTags {
+		value := tag.Value(row)
+		if tag.Index < len(row) && len(value) > 0 {
+			buffer = append(buffer, ',')
+			buffer = append(buffer, tag.LineLabel()...)
+			buffer = append(buffer, '=')
+			buffer = append(buffer, escapeTag(value)...)
+		}
+	}
+	buffer = append(buffer, ' ')
+	fieldAdded := false
+	if t.cachedFieldName != nil && t.cachedFieldValue != nil {
+		field := t.cachedFieldName.Value(row)
+		value := t.cachedFieldValue.Value(row)
+		if len(value) > 0 && len(field) > 0 {
+			buffer = append(buffer, escapeTag(field)...)
+			buffer = append(buffer, '=')
+			var err error
+			buffer, err = appendConverted(buffer, value, t.cachedFieldValue)
+			if err != nil {
+				return buffer, CsvColumnError{
+					t.cachedFieldName.Label,
+					err,
+				}
+			}
+			fieldAdded = true
+		}
+	}
+	for _, field := range t.cachedFields {
+		value := field.Value(row)
+		if len(value) > 0 {
+			if !fieldAdded {
+				fieldAdded = true
+			} else {
+				buffer = append(buffer, ',')
+			}
+			buffer = append(buffer, field.LineLabel()...)
+			buffer = append(buffer, '=')
+			var err error
+			buffer, err = appendConverted(buffer, value, &field)
+			if err != nil {
+				return buffer, CsvColumnError{
+					field.Label,
+					err,
+				}
+			}
+		}
+	}
+	if !fieldAdded {
+		return buffer, errors.New("no field data found")
+	}
+
+	if t.cachedTime != nil && t.cachedTime.Index < len(row) {
+		timeVal := t.cachedTime.Value(row)
+		if len(timeVal) > 0 {
+			if len(t.cachedTime.DataType) == 0 {
+				// assume dateTime data type (number or RFC3339)
+				t.cachedTime.DataType = dateTimeDatatype
+				t.cachedTime.DataFormat = ""
+			}
+			buffer = append(buffer, ' ')
+			var err error
+			buffer, err = appendConverted(buffer, timeVal, t.cachedTime)
+			if err != nil {
+				return buffer, CsvColumnError{
+					t.cachedTime.Label,
+					err,
+				}
+			}
+		}
+	}
+	return buffer, nil
```

Comment:
If you leverage the `bytes.Buffer` you can then return your local copy `Bytes()`

Reply:
yes, or I could simplify the function signature to use `*bytes.Buffer`